readium/shared/src/main/java/org/readium/r2/shared/resource/BytesResource.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2023 Readium Foundation. All rights reserved.
+ * Use of this source code is governed by the BSD-style license
+ * available in the top-level LICENSE file of the project.
+ */
+
+package org.readium.r2.shared.resource
+
+import kotlinx.coroutines.runBlocking
+import org.readium.r2.shared.error.Try
+import org.readium.r2.shared.extensions.coerceIn
+import org.readium.r2.shared.extensions.requireLengthFitInt
+
+sealed class BaseBytesResource(
```

Comment:
It's been sealed since long to prevent subclassing. I thought it as an internal tool, not a public API. Apps can manage on their own if there's a need. The implementation is so trivial that I'm sure it still makes sense even internally nowadays.

Reply:
Mmm, I confused this class with the `Fetcher.Resource` one. Actually,I just applied the same pattern.

Reply:
I'm fine keeping this as an internal tool, I'm just a bit confused about the sealed class. If it's just to prevent subclassing, unless we add `open` it's already the case?

Reply:
We need to subclass internally. Sealing a class is the way to enable only internal subclassing.